pkg/k8s/deployments/crud.go

```diff
@@ -132,6 +132,9 @@ func GetByDev(ctx context.Context, dev *model.Dev, namespace string, c kubernete
 			validDeployments = append(validDeployments, &dList.Items[i])
 		}
 	}
+	if len(validDeployments) == 0 {
```

Comment:
Can we test this new scenario? I think it's not covered in the tests that you added